crates/goose/src/agents/agent.rs

```diff
@@ -152,6 +155,7 @@ impl Agent {
             router_tool_selector: Mutex::new(None),
             scheduler_service: Mutex::new(None),
             retry_manager,
+            token_tracker: create_shared_tracker(),
```

Comment:
slightly worried about keeping state here - not sure why but something to think about